pkg/pillar/cmd/nim/nim.go

```diff
@@ -402,19 +402,6 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 	// We start assuming cloud connectivity works
 	dnc.CloudConnectivityWorks = true
 
-	dnc.NetworkTestBetterInterval = nimCtx.globalConfig.GlobalValueInt(types.NetworkTestBetterInterval)
-	if dnc.NetworkTestBetterInterval == 0 {
-		log.Warnln("NOT running TestBetterTimer")
-		// Dummy which is stopped needed for select loop
-		networkTestBetterTimer := time.NewTimer(time.Hour)
-		networkTestBetterTimer.Stop()
-		dnc.NetworkTestBetterTimer = networkTestBetterTimer
-	} else {
```

Comment:
Did you test this removal with the timer at zero?

In that case it isn't clear to me that handleGlobalConfigModify would create a NewTimer, hence you'll trip on a null pointer in the select loop where it does
		case _, ok := <-dnc.NetworkTestBetterTimer.C:

Reply:
This is taken care inside handleGlobalConfigModify

Reply:
As I read that it is only handling changes. If it zero how are you sure that would be detected as a change in handleGlobalConfigModify.